commit ce83f38a8a51c3abe1291878420627343cf6b4a4
Author: Marcos Mendoza <mmendoza@netgate.com>
Date:   Wed Nov 1 13:27:57 2023 -0600

    Run rc.local on system boot instead of on user logon. Fix #10980

diff --git a/src/etc/pfSense-rc b/src/etc/pfSense-rc
index 8ca5a5648e..3c995e2deb 100755
--- a/src/etc/pfSense-rc
+++ b/src/etc/pfSense-rc
@@ -576,4 +576,20 @@ echo "Bootup complete" | /usr/bin/logger
 # Reset the cache.  read-only requires this.
 /bin/rm -f /tmp/config.cache
 
+# Run /etc/rc.local
+if [ -f /etc/rc.local ]; then
+	unset do_sleep
+	if ! /bin/pgrep -qf 'rc.local$'; then
+		echo ">>> Launching rc.local in background..."
+		/bin/sh /etc/rc.local &
+		do_sleep=1
+	fi
+	if [ -f /etc/rc.local.running ] &&
+	! /bin/pgrep -qf 'rc.local.running$'; then
+		[ -n "${do_sleep}" ] && sleep 1
+		echo ">>> Launching rc.local.running in background..."
+		/bin/sh /etc/rc.local.running &
+	fi
+fi
+
 exit 0
diff --git a/src/etc/rc.initial b/src/etc/rc.initial
index e32fac0213..2dce772ce3 100755
--- a/src/etc/rc.initial
+++ b/src/etc/rc.initial
@@ -29,21 +29,6 @@ trap : INT
 trap : QUIT
 trap : ILL
 
-unset do_sleep
-if [ -f /etc/rc.local ]; then
-	if ! /bin/pgrep -qf 'rc.local$'; then
-		echo ">>> Launching rc.local in background..."
-		/bin/sh /etc/rc.local &
-		do_sleep=1
-	fi
-	if [ -f /etc/rc.local.running ] &&
-	! /bin/pgrep -qf 'rc.local.running$'; then
-		[ -n "${do_sleep}" ] && sleep 1
-		echo ">>> Launching rc.local.running in background..."
-		/bin/sh /etc/rc.local.running &
-	fi
-fi
-
 # Parse command line parameters
 while [ $# -gt 0 ]; do
 	case $1 in
